### PR TITLE
Adding requestIdFactory option

### PR DESCRIPTION
**Pull request by gabriel-pinheiro**
Allows specifying a new option: requestIdFactory
Defaults to the previous uuidv1

Closes https://github.com/puzpuzpuz/cls-rtracer/issues/37

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -295,6 +295,9 @@ These are the available config options for the middleware/plugin functions. All
   // Request header name, case insensitive (default: 'X-Request-Id').
   // Used if useHeader is set to true.
   headerName: 'X-Request-Id',
+  // Optionally choose a custom function to generate your request ids, defaults to UUIDs v1
```
Comment (puzpuzpuz) [inline README.md:298]:
```suggestion:+1
  // A custom function to generate your request ids. Defaults to UUID v1.
  // Ignored if useHeader is set to true.
```

File: index.d.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { IncomingMessage, ServerResponse } from 'http'
 
+export type RequestIdFactory = () => string
```
Comment (puzpuzpuz) [inline index.d.ts:3]:
```suggestion
export type RequestIdFactory = () => unknown
```

File: src/rtracer.js [hunk 3]
```diff
@@ -22,17 +22,20 @@ const wrapHttpEmitters = (req, res) => {
  *                                    (default: `false`)
  * @param {string} options.headerName request header name, used if `useHeader` is set to `true`
  *                                    (default: `X-Request-Id`)
+ * @param {() => string} options.requestIdFactory function used to generate request ids
```
Comment (puzpuzpuz) [inline src/rtracer.js:25]:
```suggestion
 * @param {function} options.requestIdFactory function used to generate request ids
```

File: src/rtracer.js [hunk 4]
```diff
@@ -51,12 +54,15 @@ const expressMiddleware = ({
  *                                    (default: `X-Request-Id`)
  * @param {boolean} options.useFastifyRequestId respect Fastify request id flag
  *                                    (default: `false`)
+ * @param {() => string} options.requestIdFactory function used to generate request ids
```
Comment (puzpuzpuz) [inline src/rtracer.js:57]:
```suggestion
 * @param {function} options.requestIdFactory function used to generate request ids
```

File: src/rtracer.js [hunk 5]
```diff
@@ -88,17 +94,20 @@ fastifyPlugin[Symbol.for('fastify.display-name')] = pluginName
  *                                    (default: `false`)
  * @param {string} options.headerName request header name, used if `useHeader` is set to `true`
  *                                    (default: `X-Request-Id`)
+ * @param {() => string} options.requestIdFactory function used to generate request ids
```
Comment (puzpuzpuz) [inline src/rtracer.js:97]:
```suggestion
 * @param {function} options.requestIdFactory function used to generate request ids
```

File: src/rtracer.js [hunk 6]
```diff
@@ -115,17 +124,20 @@ const koaMiddleware = ({
...
```
Comment (puzpuzpuz) [inline src/rtracer.js:127]:
```suggestion
 * @param {function} options.requestIdFactory function used to generate request ids
```

**Review by puzpuzpuz — COMMENTED** [review]
Linter fails on this PR. Could you run `npm run lint && npm run test` and fix issues?

**Comment by gabriel-pinheiro** [comment]
Nice suggestions @puzpuzpuz ! I pushed them and the linting fix (oops, forgot about it last time :X) in the last commit

**Comment by puzpuzpuz** [comment]
@gabriel-pinheiro many thanks for your contribution!

**Pull request merged**